client_cli/pytests/test/domains/test_register_domains.py

```diff
@@ -0,0 +1,103 @@
+import allure
+import pytest
+
+from common.consts import Stderr
+from src.client_cli import client_cli, have, iroha
+
+
+@pytest.fixture(scope="function", autouse=True)
+def story_account_registers_domain():
+    allure.dynamic.story('Account registers a domain')
+    allure.dynamic.label('permission', 'no_permission_required')
+
+@allure.label('sdk_test_id', 'register_domain')
+def test_register_domain(
+        GIVEN_fake_name):
+    with allure.step(
+            f'WHEN client_cli registers the domain name "{GIVEN_fake_name}"'):
+        client_cli.register().domain(GIVEN_fake_name)
+    with allure.step(
+            f'THEN Iroha should have the domain name "{GIVEN_fake_name}"'):
+        iroha.should(have.domain(GIVEN_fake_name))
```

Comment:
I agree with @mversic point.
We might consider to have as thin wrapper as possible around `client_cli`.

Reply:
Given our plans to make client CLI and all other SDKs dynamically link against a single shared object, this would become an antipattern. The API should be closely replicated in another language. Syntactic sugar and abstractions sit one layer above that.

Reply:
However, I think that this could be done as a separate ticket.